stl/src/filesys.cpp

```diff
@@ -416,17 +416,23 @@ _FS_DLL int __CLRCALL_PURE_OR_CDECL _Rename(const wchar_t* _Fname1, const wchar_
 }
 
 _FS_DLL int __CLRCALL_PURE_OR_CDECL _Resize(const wchar_t* _Fname, uintmax_t _Newsize) { // change file size
-    bool _Ok = false;
 
     HANDLE _Handle = _FilesysOpenFile(_Fname, FILE_GENERIC_WRITE, 0);
 
-    if (_Handle != INVALID_HANDLE_VALUE) { // set file pointer to new size and trim
-        LARGE_INTEGER _Large;
-        _Large.QuadPart = _Newsize;
-        _Ok             = SetFilePointerEx(_Handle, _Large, nullptr, FILE_BEGIN) != 0 && SetEndOfFile(_Handle) != 0;
-        CloseHandle(_Handle);
+    if (_Handle == INVALID_HANDLE_VALUE) {
+        return GetLastError();
     }
-    return _Ok ? 0 : GetLastError();
+
+    // set end of file information to new size
+    LARGE_INTEGER _Large;
+    _Large.QuadPart = _Newsize;
+    FILE_END_OF_FILE_INFO fileInfo;
+    fileInfo.EndOfFile = _Large;

```

Comment:
Can directly assign `fileInfo.EndOfFile.QuadPart`, thus reduce by couple of lines.
Ditto in the other file.